cEP-0011.md

```diff
@@ -0,0 +1,66 @@
+
+This document also aims at the implementation of a coafile bot, effective bear search mechanism on coala.io, enhancement of projects.coala.io and coala-html.
+
+
+# Introduction
+
+Currently, the coala Online module lacks support for all the bears. Bears with non-optional settings are ignored. coala-html lacks search functionality, which users need to filter results. The configuration of projects page is tough and requires experience of JavaScript, which prevents users to use it.
+
+This proposal focusses on enhancing coala-html by introducing results and bear search and filtering on basis of parameters. The next thing where the project will help us is that it will provide an effective bear search mechanism on coala.io and make projects.coala.io configurable and extensible. The capabilities of coala-quickstart can be used to help users generate a coafile for their project in an instant, and directly on GitHub, using a coafile bot.
+
+
+
+# Enhancements
+
+### Unified CSS Library
+
+To bring consistency throughout the web projects. Providing a CDN link to paste in the project.
+
+### coala Compliant Code in landing-frontend and projects
+
+Add proper sections in coafile and refactor code.
+
+### Bear Analytics API Endpoint
+
+This endpoint in webServices will return the click count of Bears.
+
+### coala Online Module
+
+The proposed flow is:
+
+- Use coala-quickstart’s which-bears running mode to get a list of bears which are suitable for that file.
+
+- Add another URL parameter called bear_settings, which will be a JSON having values of settings of the bears.
+
+- The backend will receive these settings in the request and run coala-quickstart in get_bears_settings mode and return the generated coafile as the response.
+
+- When this process completes, guide the backend to run coala on the given file with coafile sent in the request. The response will be the result of coala analysis in JSON format. The same procedure to be followed for one click linting support for GitHub repositories.
+
+### coafile Bot
+
+This bot will be triggered by a GitHub user mentioning the bot’s name on any issue of their repository.
+The bot will run coala-quickstart in non-interactive mode on that user’s repository and comment back on that issue with the generated coafile. It will also make a pull request with the coafile to their repository.
+
+### Configurable Projects Page
+
+Make projects page extensible by adding a YAML configuration file, making it easier for users to create their versions of projects page.
+
+### coala-html Enhancement
+
+Add search functionality, with various parameters on the basis of which results can be filtered, using angular’s ng-filter.
+
+Improve the UI of coala-html so that it uses components used in website, like cards using MaterializeCSS and coala CSS.
```

Comment:
@hemangsk I think it makes sense to cooperate with @nkprince007 and @fneu here and get coala-html into gitmate. Then the coafile bot could not only generate a coafile and make a PR but also give a URL to the results of the repo shown in gitmate